src/module/PaymentLogic.sol

```diff
@@ -253,4 +263,34 @@ library PaymentLogic {
         result.receivedProtocolFeeAmount =
             AssetManagement.transferAsset(params.sourceAsset, protocolFeeInSourceToken, _zkPayStorage.treasury);
     }
+
+    /// @notice Distributes payout amount among multiple recipients according to their percentages
+    /// @param addresses Array of payout addresses
+    /// @param percentages Array of payout percentages corresponding to addresses
+    /// @param totalAmount Total amount to distribute
+    /// @param payoutToken The token being distributed
+    function _distributePayouts(
+        address[] memory addresses,
+        uint32[] memory percentages,
+        uint256 totalAmount,
+        address payoutToken
+    ) internal {
+        uint256 base = totalAmount / MerchantLogic.TOTAL_PERCENTAGE;
+        uint256 residualTotal = totalAmount % MerchantLogic.TOTAL_PERCENTAGE;
+
+        uint256 residualAccumulator = 0;
+        uint256 numPayouts = addresses.length;
+
+        for (uint256 i = 0; i < numPayouts; ++i) {
+            uint256 percentage = percentages[i];
+
+            residualAccumulator += residualTotal * percentage;
+            uint256 amountToTransfer = base * percentage + (residualAccumulator / MerchantLogic.TOTAL_PERCENTAGE);
+            residualAccumulator %= MerchantLogic.TOTAL_PERCENTAGE;
+
+            if (amountToTransfer > 0) {
```

Comment:
No need for this validation cuz it's already handled within transferAsset method